src/ethereum_test_tools/common/types.py

```diff
@@ -2222,6 +2226,30 @@ def join(self, modifier: Header) -> "FixtureHeader":
                     setattr(new_fixture_header, header_field, value)
         return new_fixture_header
 
+    def verify(self, baseline: Header):
+        """
+        Produces a fixture header copy with the set values from the modifier.
+        """
+        for header_field in fields(self):
+            field_name = header_field.name
+            baseline_value = getattr(baseline, field_name)
+            if baseline_value is not None:
+                assert baseline_value is not Header.REMOVE_FIELD, "invalid baseline header"
+                value = getattr(self, field_name)
+                if baseline_value is Header.EMPTY_FIELD:
+                    assert value is None, f"invalid header field {header_field}"
+                    continue
+                metadata = header_field.metadata
+                assert metadata is not None, f"Field {field_name} has no header field metadata"
+                field_metadata = metadata.get("source")
+                assert isinstance(field_metadata, HeaderFieldSource), (
+                    f"Field {field_name} has invalid header_field " f"metadata: {field_metadata}"
+                )
```

Comment:
Maybe this is not needed as we do the same check within `collect()`.

Reply:
Removed :+1: